dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/RuleEngineValidators.java

```diff
@@ -68,4 +75,12 @@ public List<Validator<Enrollment>> getEnrollmentValidators()
             enrollmentRuleValidationHook,
             enrollmentAttributeValidator );
     }
+
+    @Override
+    public List<Validator<Event>> getEventValidators()
+    {
+        return List.of(
+            eventDataValuesValidator,
+            eventRuleValidator );
```

Comment:
I think it is important to maintain order here.
So `eventRuleValidator` should run before `eventDataValuesValidator`

Reply:
good catch! that is actually what I am trying to do 😄 will adapt